cellarium/ml/transforms/binomial_resample.py

```diff
@@ -0,0 +1,50 @@
+# Copyright Contributors to the Cellarium project.
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+import torch
+from torch import nn
+from torch.distributions import Bernoulli, Binomial, Uniform
+
+
+class BinomialResample(nn.Module):
+    """
+    Binomial resampling of gene counts.
+
+    For each count, the parameter to the binomial distribution
+    is independently and uniformly sampled according to the
+    bounding parameters, yielding the parameter matrix p_ng.
+
+    .. math::
+
+        y_{ng} = Binomial(n=x_{ng}, p=p_{ng})
+
+    Args:
+        p_binom_min:
+            Lower bound on binomial distribution parameter.
+        p_binom_max:
+            Upper bound on binomial distribution parameter.
+    """
+
+    def __init__(self, p_binom_min, p_binom_max, p_apply):
+        super().__init__()
+
+        self.p_binom_min = p_binom_min
+        self.p_binom_max = p_binom_max
+        self.p_apply = p_apply
+
+    def forward(self, x_ng: torch.Tensor) -> dict[str, torch.Tensor]:
+        """
+        Args:
+            x_ng: Gene counts.
+
+        Returns:
+            Binomially resampled gene counts.
+        """
+        p_binom_ng = Uniform(self.p_binom_min, self.p_binom_max).sample(x_ng.shape).type_as(x_ng)
```

Comment:
Are you using `.type_as` here to change the device?

`Uniform` uses the device of the `self.p_binom_min`. I think if it is of `float` type then it might be converted to `torch.Tensor` and be on CPU. Then `Uniform.sample` will sample on CPU which might be slow. In this case it might be better to convert `self.p_binom_min` to tensor yourself and cast to GPU device so that sampling step will be faster.